src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.Android.cs

```diff
@@ -1591,13 +1591,12 @@ internal void TryDetachView(View view)
 		/// </summary>
 		private void UpdateSelection(View view)
 		{
-			var selectorItem = view as SelectorItem;
-			
-			var item = XamlParent?.GetItemFromIndex(XamlParent?.IndexFromContainer(selectorItem) ?? -1);
-			var selectedItems = XamlParent?.SelectedItems;
-
-			if(item != null)
+			if (view is SelectorItem selectorItem &&
+				XamlParent?.IndexFromContainer(selectorItem) is int index &&
+				index != -1 &&
+				XamlParent.GetItemFromIndex(index) is object item)
```

Comment:
You should add a comment to explain this logic.

Reply:
Agree, mainly highlighting the fact we only want to execute if the `view` is a `SelectorItem`, as its not obvious that it could be a GroupHeader

Reply:
https://github.com/nventive/Uno/blob/f0e19c0467adfadf6210763caaf8e59e25ccfa89/src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.Android.cs#L1594